language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/inlayhint/InlayHintProvider.java

```diff
@@ -0,0 +1,225 @@
+import java.util.Collection;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+/**
+ * Represents the Inlay Hint Provider.
+ *
+ * @since 2201.7.0
+ */
+public class InlayHintProvider {
+    public static List<InlayHint> getInlayHint(InlayHintContext context) {
+        if (context.currentDocument().isEmpty()) {
+            return Collections.emptyList();
+        }
+        InvokableNodeFinder invokableNodeFinder = new InvokableNodeFinder();
+        Node rootNode = context.currentDocument().get().syntaxTree().rootNode();
+        rootNode.accept(invokableNodeFinder);
+
+        // Get the method name list
+        List<NonTerminalNode> invokableNodeList = invokableNodeFinder.getInvokableNodeList();
+        List<InlayHint> inlayHints = new ArrayList<>();
+
+        for (NonTerminalNode node : invokableNodeList) {
+            InlayHintArgumentTypeFinder argumentTypeFinder = new InlayHintArgumentTypeFinder();
+            node.accept(argumentTypeFinder);
+
+            // Get the argument list
+            List<NonTerminalNode> argList = argumentTypeFinder.getArgumentList();
+            if (argList.isEmpty()) {
+                return Collections.emptyList();
+            }
+            List<LineRange> commaList = argumentTypeFinder.getCommaList();
+            List<ParameterSymbol> parameterSymbols = getParameterSymbols(context, node);
+
+            for (int i = 0; i < commaList.size(); i++) {
+                LineRange lineRange = commaList.get(i);
+                int startLine = lineRange.endLine().line();
+                int startChar = lineRange.endLine().offset();
+
+                Position position = new Position(startLine, startChar);
+
+                if (parameterSymbols.size() <= i || parameterSymbols.get(i).getName().isEmpty()) {
+                    continue;
+                }
+                String label = parameterSymbols.get(i).getName().get();
+
+                InlayHint inlayHint = new InlayHint(position, Either.forLeft(label + ": "));
+                inlayHints.add(inlayHint);
+            }
+        }
+        return inlayHints;
+    }
+
+    private static List<ParameterSymbol> getParameterSymbols(InlayHintContext context,
+                                                             NonTerminalNode node) {
+        if (node.kind() == SyntaxKind.METHOD_CALL) {
+            MethodCallExpressionNode methodCallExpressionNode = (MethodCallExpressionNode) node;
+            Optional<FunctionTypeSymbol> libFunctions = TypeResolverUtil
+                    .findMethodInLangLibFunctions(methodCallExpressionNode, context);
+            if (libFunctions.isEmpty() || libFunctions.get().params().isEmpty()) {
+                return Collections.emptyList();
+            }
+            // Since the method is a lang-lib function, skip the first parameter
+            return libFunctions.get().params().get().stream().skip(1).collect(Collectors.toList());
```

Comment:
This will not work for cases like.
```
var result = regexp:find(re `John`, "John");
```
Add tests hints given for langlibs and function calls defined in other modules.

Reply:
Fixed in [e12aa9c](https://github.com/ballerina-platform/ballerina-lang/pull/40306/commits/e12aa9ce8732434b81fd4146ee59ea6f53666f49)